core/src/main/java/com/netflix/conductor/core/execution/tasks/DoWhile.java

```diff
@@ -65,15 +66,15 @@ public boolean execute(Workflow workflow, Task task, WorkflowExecutor workflowEx
 		StringBuilder failureReason = new StringBuilder();
 		Map<String, Object> output = new HashMap<>();
 		task.getOutputData().put("iteration", task.getIteration());
-		List<Task> loopOver = workflow.getTasks().stream().filter(t -> (t.getReferenceTaskName().endsWith(TaskUtils.getLoopOverTaskRefNameSuffix(task.getIteration()))) && t.isLoopOverTask()).collect(Collectors.toList());
+		List<Task> loopOver = workflow.getTasks().stream().filter(t -> (task.getWorkflowTask().has(TaskUtils.removeIterationFromTaskRefName(t.getReferenceTaskName())) && !task.getReferenceTaskName().equals(t.getReferenceTaskName()))).collect(Collectors.toList());
 
 		for (Task loopOverTask : loopOver) {
 			Status taskStatus = loopOverTask.getStatus();
 			hasFailures = !taskStatus.isSuccessful();
 			if (hasFailures) {
 				failureReason.append(loopOverTask.getReasonForIncompletion()).append(" ");
 			}
-			output.put(loopOverTask.getReferenceTaskName(), loopOverTask.getOutputData());
+			output.put(TaskUtils.removeIterationFromTaskRefName(loopOverTask.getReferenceTaskName()), loopOverTask.getOutputData());
```

Comment:
This is just code refactoring to fill DO_WHILE task output map to preserver original taskReferneceName.